register.html

```diff
@@ -35,28 +35,30 @@
                                 alt="lazy sloth logo">
                         </div>
 
-                        <form class="d-flex flex-column gap-2">
-                            <div>
+                        <form class="d-flex flex-column gap-2" id="form">
 
+                            <div>
                                 <div class="form-floating mb-3">
-                                    <input type="email" class="form-control" id="userEmail" aria-describedby="emailHelp"
-                                        placeholder="Email" required>
+                                    <input name="useremail" type="email" class="form-control" id="userEmail"
+                                        placeholder="Email" required
+                                        pattern="[\w\.%+-]+@(stud\.){0,1}noroff\.no">
                                     <label for="userEmail">Email</label>
-                                    <div class="errorMessage"></div>
+                                    <p id="regInfoText" class="text-black opacity-50 subText ms-2">Only noroff.no emails are allowed to register</p>
                                 </div>
 
                                 <div class="form-floating mb-3">
-                                    <input type="text" class="form-control" id="userName" placeholder="Username"
-                                        required pattern="^[A-Za-z][A-Za-z0-9_]{3,29}$">
+                                    <input name="username" type="text" class="form-control" id="userName"
+                                        placeholder="Username" required pattern="^[A-Za-z][A-Za-z0-9_]{3,29}$">
                                     <label for="userName">Username</label>
-                                    <div class="errorMessage"></div>
+                                    <p class="text-black opacity-50 subText ms-2">Name can only use a-Z, 0-9, and _</p>
                                 </div>
 
                                 <div class="form-floating mb-3">
-                                    <input type="password" id="userPassword" class="form-control" placeholder="Password"
-                                        required autocomplete="new-password">
+                                    <input name="password" type="password" id="userPassword" class="form-control"
+                                        placeholder="Password" required autocomplete="new-password"
+                                        pattern="/^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[a-zA-Z]).{8,}$/gm">
```

Comment:
```suggestion
                                        minlength="8">
```

Restricting password character choices is an anti-pattern that makes applications less, rather than more, secure.

The more you restrict the user's character choice, the more information you provide to a hacker about where to start guessing. Mystery (obfuscation) is a good thing here.